scalyr_agent/platform_windows.py

```diff
@@ -896,7 +926,11 @@ def close(self):
                 self.__pipe_handle = None
 
 
-if __name__ == "__main__":
+# Workaround to make sure we don't run this as part of tests on import since it breaks
+# things
+if __name__ == "__main__" and not not compat.os_environ_unicode.get(
+    "INSIDE_WINDOWS_TESTS_PYTEST", False
```

Comment:
Will this have an effect when the `platform_windows` module is imported from somewhere else and is not an "entry point" for the interpreter?

Reply:
It shouldn't - I tried to design it so the second part of if will only be False during tests (that's also the reason why I tried to a use a unique environment variable name and not something just like ``TEST`` which should never appear in real-life)

The reason for that is that this module has side effect on import which breaks things which means we can't unit test it without a workaround like that.

Reply:
Managed to change tests up a bit so we don't need this workaround anymore - now we run platform_windows tests in a standalone fashion.